File: frontend_flutter/lib/domain/entities/day_closure.dart
class DayClosure {
  final String id;
  final String date;
  final String status;
  final String closedBy;
  final String closedAt;
  final String notes;
  final List<Map<String, dynamic>> reopenLog;

  DayClosure({
    required this.id,
    required this.date,
    required this.status,
    required this.closedBy,
    String? closedAt,
    this.notes = '',
    List<Map<String, dynamic>>? reopenLog,
  })  : closedAt = closedAt ?? DateTime.now().toUtc().toIso8601String(),
        reopenLog = reopenLog ?? [];

  factory DayClosure.fromJson(Map<String, dynamic> json) {
    return DayClosure(
      id: json['id'] ?? '',
      date: json['date'] ?? '',
      status: json['status'] ?? 'ABIERTO',
      closedBy: json['closed_by'] ?? '',
      closedAt: json['closed_at'],
      notes: json['notes'] ?? '',
      reopenLog: (json['reopen_log'] as List?)?.cast<Map<String, dynamic>>(),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'date': date,
    'status': status,
    'closed_by': closedBy,
    'closed_at': closedAt,
    'notes': notes,
    'reopen_log': reopenLog,
  };
}

File: packages/backend/app/models/day_closure.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import DateTime, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class DayClosure(Base):
    __tablename__ = "day_closures"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    date: Mapped[str] = mapped_column(String(10), unique=True, index=True)
    status: Mapped[str] = mapped_column(String(30))
    closed_by: Mapped[str | None] = mapped_column(
        String(36), ForeignKey("usuarios.id")
    )
    closed_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )
    notes: Mapped[str | None] = mapped_column(Text)
    reopen_log_json: Mapped[str] = mapped_column(Text, default="[]")
